### PR TITLE
Allow file paths in `filename` attribute

### PR DESCRIPTION
**Pull request by donquxiote**
I wanted to use this action but needed to reference a file that's a few directories deep. The Github Gist API does not allow separators like "/" in the `name` attribute in the `files` parameter.

This adds a check for "/", and then based on that passes the file name without separators to the Gist API.

Also added a tool-versions file since I use mise instead of nvm.

### CODE REVIEW

File: src/syncGist.test.js [hunk 1]
```diff
@@ -19,6 +21,18 @@ test('create and delete gist', async () => {
   ).resolves.not.toThrow();
 });
 
+test('create and delete gist from nested file', async () => {
+  await ensureDir(directoryName);
+  await writeFile(nestedFilename, defaultFileData);
+  const { id } = await syncGist(gistPat, '', 'create', filename);
```
Comment (danielmcconville) [inline src/syncGist.test.js]: Does not test the `nestedFilename`


```suggestion
  const { id } = await syncGist(gistPat, '', 'create', nestedFilename);
```

File: src/syncGist.test.js [hunk 2]
```diff
@@ -19,6 +21,18 @@ test('create and delete gist', async () => {
...
+  expect(id).toBeDefined();
+  await expect(
+    syncGist(gistPat, id, 'delete', filename)
```
Comment (danielmcconville) [inline src/syncGist.test.js]:
```suggestion
    syncGist(gistPat, id, 'delete', nestedFilename)
```

File: src/syncGist.js [hunk 3]
```diff
@@ -13,12 +13,18 @@ const syncGist = async (
     auth,
   });
 
+  var parsedFileName = filename
+
+  if (filename.includes("/")) {
+     parsedFileName = filename.split("/").at(-1)
```
Comment (danielmcconville) [inline src/syncGist.js]: Better to use `path` to handle this so edge cases are caught (trailing slashes, multiple slashes etc).


```suggestion
     parsedFileName = path.basename(filename);
```

File: src/syncGist.js [hunk 4]
```diff
@@ -13,12 +13,18 @@ const syncGist = async (
     auth,
   });
 
+  var parsedFileName = filename
```
Comment (danielmcconville) [inline src/syncGist.js]: Ideally, stick to using `let` and `;` at the end of the line for consistency

```suggestion
  let parsedFileName = filename;
```

File: src/syncGist.test.js [hunk 5]
```diff
@@ -63,3 +77,32 @@ test('create then update to gist, then delete', async () => {
...
+async function ensureDir(path) {
+  try {
+    await mkdir(path, { recursive: true });
+  } catch (err) {
+    // ignore error if the directory already exists
+    if (err.code !== 'EEXIST') {
+      throw err;
+    }
+  }
+}
```
Comment (danielmcconville) [inline src/syncGist.test.js]: `mkdir(path, { recursive: true })` should already handle existing directories without throwing so you could remove this function and just call this inline.

**Review by danielmcconville — CHANGES_REQUESTED** [review]
Thanks for the PR! Supporting file paths in the filename input is a useful addition.

A few things to address before merging:

1. Use path.basename() instead of manual string splitting - filename.split("/").at(-1) doesn't handle Windows backslashes
  or edge cases. Node's built-in path.basename() handles all of this and reduces the block to a single line.
2. Test bug in "create and delete gist from nested file" - Lines 27 and 30 pass filename instead of nestedFilename, so this
   test doesn't actually exercise the new path-stripping logic.
3. var → const, missing semicolons, 3-space indent - Minor style issues, but they'll be resolved naturally by the
  path.basename() change.

See inline comments for details.

**Comment by donquxiote** [comment]
@danielmcconville Thanks for the help! I'm usually working in Ruby and it's been a while since I touched raw javascript, so apologies for the minor bugs and style things.

**Review by danielmcconville — APPROVED** [review]
@donquxiote - no problem, thanks for the PR!

I added a few tweaks myself.
